plugin/src/main/java/org/opensearch/ml/rest/RestMLTaskProfileAction.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.rest;
+
+import static org.opensearch.ml.plugin.MachineLearningPlugin.ML_BASE_URI;
+import static org.opensearch.ml.utils.RestActionUtils.PARAMETER_TASK_ID;
+import static org.opensearch.ml.utils.RestActionUtils.getAllNodes;
+import static org.opensearch.ml.utils.RestActionUtils.getParameterId;
+import static org.opensearch.ml.utils.RestActionUtils.onFailure;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import lombok.extern.log4j.Log4j2;
+
+import org.opensearch.action.ActionListener;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.xcontent.ToXContent;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.ml.action.profile.MLTaskProfileAction;
+import org.opensearch.ml.action.profile.MLTaskProfileNodeResponse;
+import org.opensearch.ml.action.profile.MLTaskProfileRequest;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.BytesRestResponse;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.RestStatus;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+
+@Log4j2
+public class RestMLTaskProfileAction extends BaseRestHandler {
```

Comment:
Going to create multiple profile APIs, like one for task profile, another one for model? How about we keep one profile API and user can specify the target they want to detect?

Reply:
I understand your point. We can refactor this one and let user to define the target in the request body.